pytensor/tensor/rewriting/math.py

```diff
@@ -1905,12 +1905,27 @@ def local_reciprocal_canon(fgraph, node):
 @register_canonicalize
 @node_rewriter([pt_pow])
 def local_pow_canonicalize(fgraph, node):
-    cst = get_underlying_scalar_constant_value(
+    """
+    Rewrites for exponential functions with straight-forward simplifications:
+    1. x ** 0 -> 1
+    2. x ** 1 -> x
+    3. 1 ** x -> 1
+
+    In all cases, the shape of the output is the result of broadcasting the shapes of the inputs.
+    """
+
+    cst_base = get_underlying_scalar_constant_value(
+        node.inputs[0], only_process_constants=True, raise_not_constant=False
+    )
+    if cst_base == 1:
+        return [broadcast_arrays(*node.inputs)[0].astype(node.outputs[0].dtype)]
```

Comment:
Can clean up a bit by storing `inp_idx` in the 3 branches and doing something like:

```
inp_idx = None
if case1:
  inp_idx = 1
elif case 2
  inp_idx = 0
...

if inp_idx is None:
  return None

new_out = broadcast_arrays(*node.inputs)[inp_idx]

if new out.dtype != node.out.dtype:
  new_out = cast(...)

return [new_out]
```
?

Reply:
my concern about alloc_like applies to the old code as well

Reply:
I don't think it can be as clean as this, because in the x ** 0 case we're not using either input value in the output. But I tried to make it look more like this.